pkg/cronjob/es_index_cleaner_test.go

```diff
@@ -39,6 +40,38 @@ func TestCreateEsIndexCleaner(t *testing.T) {
 	assert.Equal(t, historyLimits, *cronJob.Spec.SuccessfulJobsHistoryLimit)
 }
 
+func TestCreateEsIndexCleanerTypeMeta(t *testing.T) {
+	testData := []struct {
+		Name string
+		flag string
+	}{
+		{Name: "Test batch/v1beta1", flag: v1.FlagCronJobsVersionBatchV1Beta1},
+		{Name: "Test batch/v1", flag: v1.FlagCronJobsVersionBatchV1},
+	}
+
+	jaeger := &v1.Jaeger{Spec: v1.JaegerSpec{Storage: v1.JaegerStorageSpec{Options: v1.NewOptions(
+		map[string]interface{}{"es.index-prefix": "tenant1", "es.server-urls": "http://nowhere:666,foo"})}}}
+	days := 0
+	jaeger.Spec.Storage.EsIndexCleaner.NumberOfDays = &days
+	historyLimits := int32(1)
+	jaeger.Spec.Storage.EsIndexCleaner.SuccessfulJobsHistoryLimit = &historyLimits
+	for _, td := range testData {
+		if td.flag == v1.FlagCronJobsVersionBatchV1Beta1 {
+			viper.SetDefault(v1.FlagCronJobsVersion, v1.FlagCronJobsVersionBatchV1Beta1)
+		}
+		cronJobs := CreateEsIndexCleaner(jaeger)
+		switch tt := cronJobs.(type) {
+		case *batchv1beta1.CronJob:
+			assert.Equal(t, tt.Kind, "CronJob")
```

Comment:
Expected outcome can also be a part of testCase object. It helps when there are multiple test cases, instead of using a hard-coded value, you can simply pick it from the testcase like:
``` assert.Equal(t, tt.Kind, t.ExpectedKind) ```